auction-server/src/auction/service/verification.rs

```diff
@@ -533,82 +529,206 @@ impl Service<Svm> {
+                    ),
+                    QuoteTokens::OutputTokenSpecified {
+                        input_token,
+                        output_token,
+                        ..
+                    } => (
+                        swap_data.amount_input,
+                        input_token,
+                        None,
+                        output_token.token,
+                        Some(output_token.amount),
+                    ),
+                };
+                if let Some(expected_input_amount) = expected_input_amount {
+                    if expected_input_amount != swap_data.amount_input {
+                        return Err(RestError::BadParameters(
+                            format!(
+                                "Input amount in swap opportunity {} does not match the input amount in the swap instruction {}",
+                                expected_input_amount, swap_data.amount_input
+                            ),
+                        ));
+                    }
+                }
+                if let Some(expected_output_amount) = expected_output_amount {
+                    if expected_output_amount != swap_data.amount_output {
+                        return Err(RestError::BadParameters(
+                            format!(
+                                "Output amount in swap opportunity {} does not match the output amount in the swap instruction {}",
+                                expected_output_amount, swap_data.amount_output
+                            ),
+                        ));
+                    }
+                }
+                if expected_input_token != mint_input {
+                    return Err(RestError::BadParameters(
+                        format!(
+                            "Input token in swap opportunity {} does not match the input token in the swap instruction {}",
+                            expected_input_token, mint_input
+                        ),
+                    ));
+                }
+                if expected_output_token != mint_output {
+                    return Err(RestError::BadParameters(
+                        format!(
+                            "Output token in swap opportunity {} does not match the output token in the swap instruction {}",
+                            expected_output_token, mint_output
+                        ),
+                    ));
+                }
+
+
+                if swap_data.referral_fee_bps != opp_swap_data.referral_fee_bps {
+                    return Err(RestError::BadParameters(
+                        format!(
+                            "Referral fee bps in swap opportunity {} does not match the referral fee bps in the swap instruction {}",
+                            opp_swap_data.referral_fee_bps, swap_data.referral_fee_bps
+                        ),
+                    ));
+                }
+                if user_wallet != opp_swap_data.user_wallet_address {
```

Comment:
this is probably the first one i would show, before referral fee and token mints and amounts